src/main/java/com/algorand/sdkutils/listeners/JavaGenerator.java

```diff
@@ -118,6 +118,9 @@ public void onEvent(Events event, TypeDef type) {
         case BODY_CONTENT:
             javaQueryWriter.addQueryProperty(type, false, false, true);
             break;
+        case ENUM_DEFINITION:
+            this.storeEnumDefinition(type);
```

Comment:
On closer inspection, I think we can add this check with the current parser by adding this to the TypeDef event:
```diff
         case NEW_PROPERTY:
             javaModelWriter.newProperty(type);
+            if (type.enumValues != null && type.enumValues.size() > 0) {
+                 this.storeEnumDefinition(type);
+            }
```

Reply:
I don't think this will work. The reason we have to change the parser is because `generateAlgodIndexerObjects` is the only method which is looking at the `definitions` block, and the following code block ensure that objects in that block are skipped if they don't have `properties` defined, which enums do not have.
```
if (!hasProperties(cls.getValue())) {
    // If it has no properties, no class is needed for this type.
    continue;
}
```

So in that case we wouldn't ever enter the `NEW_PROPERTY` event block.

Reply:
Also, as an aside, enums implicity defined in the Parameters section are omitted from the generated code (Format)